python/python/tests/test_async_client.py

```diff
@@ -718,6 +718,20 @@ async def test_hexist(self, redis_client: TRedisClient):
         assert await redis_client.hexists(key, "nonExistingField") == False
         assert await redis_client.hexists("nonExistingKey", field2) == False
 
+    @pytest.mark.parametrize("cluster_mode", [True, False])
+    @pytest.mark.parametrize("protocol", [ProtocolVersion.RESP2, ProtocolVersion.RESP3])
+    async def test_hlen(self, redis_client: TRedisClient):
+        key = get_random_string(10)
+        field = get_random_string(5)
+        field2 = get_random_string(5)
+        field_value_map = {field: "value", field2: "value2"}
+
+        assert await redis_client.hset(key, field_value_map) == 2
+        assert await redis_client.hlen(key) == 2
+        assert await redis_client.hdel(key, [field]) == 1
+        assert await redis_client.hlen(key) == 1
+        assert await redis_client.hlen("non_existing_hash") == 0
```

Comment:
please add a pytest.raises test for a key that holds different type

Reply:
done